### PR TITLE
Allow selective sync of spaces in folder wizard

### PR DESCRIPTION
**Pull request by fmoc**
Closes https://github.com/owncloud/client/issues/10596.

### CODE REVIEW

File: src/gui/selectivesyncwidget.cpp [hunk 1]
```diff
@@ -263,8 +264,8 @@ void SelectiveSyncWidget::slotItemExpanded(QTreeWidgetItem *item)
     QString dir = item->data(0, Qt::UserRole).toString();
     if (dir.isEmpty())
         return;
-    // TODO: legacy
-    PropfindJob *job = new PropfindJob(_account, _account->davUrl(), _folderPath + dir, PropfindJob::Depth::One, this);
+    Q_ASSERT(!_davUrl.isEmpty());
```
Comment (TheOneRing) [inline src/gui/selectivesyncwidget.cpp:267]: How about a getter with that check?

Comment (fmoc) [reply src/gui/selectivesyncwidget.cpp:267]: Would save a tiny bit of code duplication, but we cannot enforce the use of that getter over the variable anyway...

Comment (TheOneRing) [reply src/gui/selectivesyncwidget.cpp:267]: There are private getters :D
And in general I'd say we should always use getters instead of raw vars.

Comment (fmoc) [reply src/gui/selectivesyncwidget.cpp:267]: Well, if you prefer it...

Comment (TheOneRing) [reply src/gui/selectivesyncwidget.cpp]: Qt codestyle does not use get as getter prefix

**Comment by TheOneRing** [comment]
Can you check whether the ignore list computes the correct patterns?
image
Hidden files should actually not appear afaik.

**Review by TheOneRing — APPROVED** [review]

**Pull request merged**